### PR TITLE
dolphin-emu-primehack: init at 1.0.5

### PR DESCRIPTION
**Pull request by Madouura**
###### Motivation for this change
Dolphin fork specifically for the Metroid Prime games.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSamus** [comment]
Passing by to say that there is a small debate about renaming `dolphinEmu` to `dolphin-emu` in #145384.

**Comment by Madouura** [comment]
> Passing by to say that there is a small debate about renaming `dolphinEmu` to `dolphin-emu` in #145384. Also, this suffers from the same problem of #145185.

If they merge, I'll go ahead and rebase to suit the new changes.

**Comment by SuperSamus** [comment]
Though, in my PR the i removed the "new" Dolphin executable and .desktop file because there is little point to have both stable and beta versions of Dolphin installed.
In this case, PrimeHack is a fork that has a completely different goal from the original, so having both Dolphin and PrimeHack installed makes sense, so this _should_ having a new executable name and .desktop (as long as the old ones are removed/hidden).
(Just a thing I was thinking, I am not the one who has the say on that)

**Comment by SuperSamus** [comment]
I don't know if GitHub sent an email, but the renaming PR has been merged.

File: pkgs/misc/emulators/dolphin-emu/primehack.nix [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+, curl
+, alsa-lib
+, miniupnpc
+, enet
+, mbedtls
+, soundtouch
+, sfml
+, fmt
+, vulkan-loader ? null
+, libpulseaudio ? null
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dolphin-emu/primehack.nix:39]:
```suggestion
, vulkan-loader
, libpulseaudio
```

those packages are under normal condition not null.

File: pkgs/misc/emulators/dolphin-emu/primehack.nix [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  postPatch = ''
+    sed -i -e 's,DISTRIBUTOR "None",DISTRIBUTOR "NixOS",g' CMakeLists.txt
+  '' + lib.optionalString stdenv.isDarwin ''
+    sed -i -e 's,if(NOT APPLE),if(true),g' CMakeLists.txt
+    sed -i -e 's,if(LIBUSB_FOUND AND NOT APPLE),if(LIBUSB_FOUND),g' \
+      CMakeLists.txt
+  '';
+
+  postInstall = ''
+    cp -r ${desktopItem}/share/applications $out/share
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dolphin-emu/primehack.nix:144]: Please use copyDesktopItems instead.

File: pkgs/misc/emulators/dolphin-emu/primehack.nix [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  '' + lib.optionalString stdenv.hostPlatform.isLinux ''
+    install -D $src/Data/51-usb-device.rules $out/etc/udev/rules.d/51-usb-device.rules
+  '';
+
+  meta = with lib; {
+    homepage = "https://dolphin-emu.org";
+    description = "Gamecube/Wii/Triforce emulator for x86_64 and ARMv8";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ MP2E ashkitten Madouura ];
+    branch = "master";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dolphin-emu/primehack.nix:155]: meta.branch is usually not set anymore.

**Comment by SuperSamus** [comment]
With this, having both PrimeHack and regular Dolphin installed makes only one of them visible.
In the [AUR version of the package](https://aur.archlinux.org/packages/dolphin-emu-primehack-git/) conflicts with regular Dolphin, but I don't think it should be the same here on nixpkgs:
> Though, in my PR I removed the "new" Dolphin executable and .desktop file because there is little point to have both stable and beta versions of Dolphin installed.
In this case, PrimeHack is a fork that has a completely different goal from the original, so having both Dolphin and PrimeHack installed makes sense, so this _should_ have a new executable name and .desktop (as long as the old ones are removed/hidden).
(Just a thing I was thinking, I am not the one who has the say on that)

**Comment by Madouura** [comment]
Did some general cleanup.

**Pull request merged**